### DISCUSSION TITLE
Can I use prisma and node-postgres tougether?

### DISCUSSION THREAD

**Discussion by jtlindsey** [prisma/prisma#9356] [Q&A] [answered]
Hello,
I started using [`prisma`][1] especially for handling database migrations. It handles that well. But there are many open issues for things that it does not handle well related to queries (biggest are related to `queryRaw` not always working as expected and with no straight forward way to use `postgreSQL` Row Level Security). Everything I've found to be a problem related to queries in `prsima` can easily be done in [`node-postgres`][2].

I realize from the docs that some of these issues are not a problem in `knexjs` but `prisma` has a more feature rich migration setup ([automatic][3] and can be [customized][4]).

Can I safely use `prisma` and `node-postgres` together? If so, how? For example use `prisma` for schema design and database migrations and use `node-postgres` for all my query logic?

I have a project where progress on the database has stopped because of issues in `prisma` that are not a problem when using `node-postgres`

  [1]: https://www.npmjs.com/package/prisma
  [2]: https://www.npmjs.com/package/pg
  [3]: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate
  [4]: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations

**Comment by TasinIshmam** [comment, marked as answer]
Hi @jtlindsey  :wave:

Yes, you can safely use ```prisma``` and ```node-postgres``` together. 

Your workflow will look like the following: 

 1. Create a  [Prisma Schema](https://www.prisma.io/docs/concepts/components/prisma-schema) to define your models and underlying database tables.
 2. Use the [Prisma CLI](https://www.prisma.io/docs/concepts/components/prisma-cli) (the ```prisma``` npm library) to run migrations to your database. This will generate the client in ```node_modules/.prisma/client``` directory, which you could optionally delete as you won't be using it. 
 3. Instead of using generated [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) inside your node application, use the ```node-postgres``` library to run queries against the database in your application. 


Since you're only using Prisma to make any kind of migration or changes to the database structure, there's no risk of your prisma schema not being synced with your database. On the off chance this does happen, you can always get it back in sync using ```prisma db pull```. 

Furthermore, since your application is not going to use Prisma client to connect to the database for running queries, ```node-postgres``` will handle the connection pool. 

Btw, I also posted this response [on your question in stackoverflow](https://stackoverflow.com/questions/69268867/can-i-use-prisma-and-node-postgres-tougether/).

**Reply by jtlindsey** [reply]
@TasinIshmam thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
